### PR TITLE
GC+: Fix list objects

### LINKED ISSUE #4683: GC+: Fix list objects

### PR DESCRIPTION
**Pull request by N-o-Z**
closes #4683

### CODE REVIEW

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 1]
```diff
@@ -52,11 +52,11 @@ object UncommittedGarbageCollector {
...
     // TODO (niro): implement parallel lister for old repositories (https://github.com/treeverse/lakeFS/issues/4620)
     val oldDataDF = new NaiveDataLister().listData(configMapper, oldDataPath)
-    dataDF.union(oldDataDF)
+    dataDF =
+      dataDF.union(oldDataDF).filter(dataDF("last_modified") < before.getTime).select("address")
 
-    dataDF = dataDF.filter(dataDF("last_modified") < before.getTime).select("address")
     // TODO (niro): Check if really needed and consider passing it as part of uncommittedGCRunInfo
-    dataDF = dataDF.filter(dataDF("address").startsWith("_lakefs"))
+    dataDF = dataDF.filter(!dataDF("address").startsWith("_lakefs"))
```
Comment (itaiad200) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:59]: I might be misunderstanding something..
How is `_lakefs\` being listed at all? I thought the `oldDataDF` will only list objects which are directly under the storage namespace. If it lists `_lakefs\` why doesn't it list `data\` too?

Comment (N-o-Z) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:59]: That's why I added the comment - anyway this is not the only issue here

Comment (itaiad200) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:59]: We possibly list everything twice, I think we should address it now (can be in a separate PR though)

Comment (N-o-Z) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:59]: We don't list everything twice - this is already tested

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 2]
```diff
@@ -49,14 +50,12 @@ object UncommittedGarbageCollector {
       .select("address", "last_modified")
 
     // Read objects from namespace root, for old structured repositories
-    val oldDataPath = new Path(storageNamespace)
+
     // TODO (niro): implement parallel lister for old repositories (https://github.com/treeverse/lakeFS/issues/4620)
     val oldDataDF = new NaiveDataLister().listData(configMapper, oldDataPath)
-    dataDF.union(oldDataDF)
+    dataDF =
+      dataDF.union(oldDataDF).filter(dataDF("last_modified") < before.getTime).select("address")
```
Comment (itaiad200) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:57]: Will the filtering work correctly? 
The `oldDataDF` isn't part of `dataDF` when the args are captured. I think the `oldDataDF` won't get filtered.

**Review by itaiad200 — CHANGES_REQUESTED** [review]

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala (hunk 2 above)
Comment (N-o-Z) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala]: This is just to provide the column name - modified to use `col` instead

**Review by itaiad200 — APPROVED** [review]
Thanks!

**Pull request merged**